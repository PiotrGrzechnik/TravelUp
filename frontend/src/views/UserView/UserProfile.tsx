import React, { useState } from 'react'
import { useDispatch } from 'react-redux'
import styled from 'styled-components'
import { Card, Avatar, Descriptions, Button } from 'antd'
import { UserOutlined } from '@ant-design/icons'

const Container = styled.div`
  display: flex;
`
const AvatarStyled = styled(Avatar)`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 70px;
  width: 70px;
  margin: auto;

  && {
    font-size: 44px;
  }
`
const CardContent = styled(Card)`
  min-height: 400px;
`

type UserProfileProps = {
  user?: {
    id?: number
    name?: string
    email?: string
  }
}

const UserProfile: React.FC<UserProfileProps> = props => {
  const [isEdit, setIsEdit] = useState(false)
  const { user } = props

  return (
    <>
      {user && (
        <Container>
          <Card title={<AvatarStyled size="large" icon={<UserOutlined />} />}>
            <p>{user.name}</p>
            <p>{user.email}</p>
          </Card>
          <CardContent
            title={
              <div>
                User Info {!isEdit && <Button onClick={() => setIsEdit(true)}>Edit</Button>}
              </div>
            }
          >
            {isEdit ? (
              <div>Edit page</div>
            ) : (
              <Descriptions>
                <Descriptions.Item label="UserName">{user.name}</Descriptions.Item>
                <Descriptions.Item label="E-mail">{user.email}</Descriptions.Item>
              </Descriptions>
            )}
          </CardContent>
        </Container>
      )}
    </>
  )
}

export default UserProfile
