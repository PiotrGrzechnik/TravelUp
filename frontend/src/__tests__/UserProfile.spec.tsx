import React from 'react'
import Enzyme, { shallow, mount, render } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'

Enzyme.configure({ adapter: new Adapter() })

import UserProfile from 'src/views/UserView/UserProfile'

describe('UserProfile', () => {
  test('renders component without crashing', () => {
    const user = {}
    const app = renderer.create(<UserProfile user={user} />).toJSON()
    expect(app).toMatchSnapshot()
  })
})
