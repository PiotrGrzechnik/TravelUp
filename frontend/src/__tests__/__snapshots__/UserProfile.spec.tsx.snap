// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile renders component without crashing 1`] = `
<div
  className="sc-AxjAm kUtHQi"
>
  <div
    className="ant-card sc-AxirZ hUzuQU ant-card-bordered"
  >
    <div
      className="ant-card-head"
      style={Object {}}
    >
      <div
        className="ant-card-head-wrapper"
      >
        <div
          className="ant-card-head-title"
        >
          <img
            className="sc-AxiKw jA-dUxj"
            src=""
          />
        </div>
      </div>
    </div>
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <p
        className="sc-AxhUy gYyJZg"
      />
      <p
        className="sc-AxhUy gYyJZg"
      />
    </div>
  </div>
  <div
    className="ant-card sc-AxhCb bKdAJu ant-card-bordered"
  >
    <div
      className="ant-card-head"
      style={Object {}}
    >
      <div
        className="ant-card-head-wrapper"
      >
        <div
          className="ant-card-head-title"
        >
          <div>
            User Info 
            <button
              className="ant-btn"
              onClick={[Function]}
              type="button"
            >
              <span>
                Edit
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <div
        className="ant-descriptions"
      >
        <div
          className="ant-descriptions-view"
        >
          <table>
            <tbody>
              <tr
                className="ant-descriptions-row"
              >
                <td
                  className="ant-descriptions-item"
                  colSpan={3}
                >
                  <span
                    className="ant-descriptions-item-content"
                  >
                    Country
                    Gender
                  </span>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;
